### ISSUE TITLE
Crashing with OBS Plugin

### ISSUE THREAD

**Issue by dv-anomaly** [dalegaard/ndiflbridge#1]
I'm posting this here, since issues are disabled at the obs-studio repo. Not sure what is at fault here.

After Compiling https://github.com/dalegaard/obs-studio

This is my output:

```
avdept@av-workstation-1 ~/b/ndiflbridge> obs
Attempted path: share/obs/obs-studio/locale/en-US.ini
Attempted path: /usr/share/obs/obs-studio/locale/en-US.ini
Attempted path: share/obs/obs-studio/locale.ini
Attempted path: /usr/share/obs/obs-studio/locale.ini
Attempted path: share/obs/obs-studio/themes/Default.qss
Attempted path: /usr/share/obs/obs-studio/themes/Default.qss
Attempted path: share/obs/obs-studio/license/gplv2.txt
Attempted path: /usr/share/obs/obs-studio/license/gplv2.txt
info: Processor: 24 logical cores
info: Processor: Intel(R) Xeon(R) CPU           X5670  @ 2.93GHz
info: Processor: Intel(R) Xeon(R) CPU           X5670  @ 2.93GHz
info: Processor: Intel(R) Xeon(R) CPU           X5670  @ 2.93GHz
info: Processor: Intel(R) Xeon(R) CPU           X5670  @ 2.93GHz
info: Physical Memory: 48285MB Total
info: Kernel Version: Linux 4.8.13-1-ARCH
info: Distribution: "Antergos Linux" Unknown
info: Portable mode: false
QObject::connect: invalid null parameter
info: OBS 0.16.2.r177.g1790470f (linux)
info: ---------------------------------
info: ---------------------------------
info: audio settings reset:
	samples per sec: 44100
	speakers:        2
info: OpenGL version: 4.3 (Core Profile) Mesa 13.0.2

info: ---------------------------------
info: video settings reset:
	base resolution:   1600x900
	output resolution: 1280x720
	downscale filter:  Bicubic
	fps:               30/1
	format:            NV12
info: ---------------------------------
libDeckLinkAPI.so: cannot open shared object file: No such file or directory
info: No blackmagic support
error: os_dlopen(libnvidia-encode.so.1->libnvidia-encode.so.1): libnvidia-encode.so.1: cannot open shared object file: No such file or directory

error: Register output
error: Spawned finder ndibridge
execve: No such file or directory
terminate called without an active exception
info: VLC found, VLC video source enabled
info: ---------------------------------
info:   Loaded Modules:
info:     vlc-video.so
info:     text-freetype2.so
info:     rtmp-services.so
info:     obs-x264.so
info:     obs-transitions.so
info:     obs-outputs.so
info:     obs-ndi.so
info:     obs-libfdk.so
info:     obs-filters.so
info:     obs-ffmpeg.so
info:     linux-v4l2.so
info:     linux-pulseaudio.so
info:     linux-jack.so
info:     linux-decklink.so
info:     linux-capture.so
info:     linux-alsa.so
info:     image-source.so
info:     frontend-tools.so
info: ==== Startup complete ===============================================
error: Service '' not found
info: All scene data cleared
info: ------------------------------------------------
info: pulse-input: Server name: 'pulseaudio 9.0'
error: pulse-input: An error occurred while getting the source info!
info: pulse-input: Server name: 'pulseaudio 9.0'
error: pulse-input: An error occurred while getting the source info!
info: Switched to scene 'Scene'
info: ------------------------------------------------
info: Loaded scenes:
info: - scene 'Scene':
info: ------------------------------------------------
error: NDI finder crashed: Error reading from ndibridge find

info: Update check: last known remote version is 0.4.0
error: Spawned finder ndibridge
execve: No such file or directory
terminate called without an active exception
error: NDI finder crashed: Error reading from ndibridge find

error: Spawned finder ndibridge
execve: No such file or directory
terminate called without an active exception
error: NDI finder crashed: Error reading from ndibridge find

error: Spawned finder ndibridge
execve: No such file or directory
terminate called without an active exception
error: NDI finder crashed: Error reading from ndibridge find

```

**Comment by dv-anomaly** [comment]
Nevermind, i think I got it, I copied ndiflbridge to `/usr/bin` instead of `/usr/local/bin`

**Comment by dv-anomaly** [comment]
I am now getting this error when adding a source:

```
info: NDI Subscribing to ADMIN-PC (Test Pattern)
error: Link name: fl-v3HUAYG1
error: Spawned receiver ndibridge pid 27926
Stream worker starting for 1 / ADMIN-PC (Test Pattern)
terminate called after throwing an instance of 'char const*'

```

```
info: NDI Subscribing to ADMIN-PC (Desktop)
error: Link name: fl-XcIUFkLk
error: Spawned receiver ndibridge pid 27847
Stream worker starting for 1 / ADMIN-PC (Desktop)
terminate called after throwing an instance of 'char const*'
```

**Comment by dalegaard** [comment]
The management of the ndiflbridge from the obs plugin is currently pretty hacked together, I've been meaning to rewrite it with [tiny-process-library](https://github.com/eidheim/tiny-process-library) but have yet to find the time. That should handle Windows compat as well.

I've not encountered your specific problem however, but I'm thinking it will be fixed along with that work as well. On my local machine everything appears to work with the current versions pushed to both repositories.

**Comment by dalegaard** [comment]
I believe I found the problem. When upgrading the NDI SDK I mistakenly forgot to handle the BGRX case - I had never observed it in the previous SDK apparently. It's fixed in 5ab5351. Can you verify and close this if the problem is fixed? You should be able to receive NDI now, but sending will still not work correctly.

**Comment by dv-anomaly** [comment]
Ok, I will be able to test that out in a couple hours. Thanks! I don't know C very well. So I was a little lost on some of the code. LOL I see what you did there pretty easy fix.
